.circleci/config.yml

```diff
@@ -20,34 +20,32 @@ commands:
     description: Install Ruby gems
     steps:
       - restore_cache:
-          keys: # Include fallbacks to increase the likeliness of a cache hit
-            - gem-cache-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
-            - gem-cache-{{ arch }}-{{ .Branch }}
-            - gem-cache-{{ arch }}-
+          keys:
+            - bundler-cache-v0-{{ arch }}-{{ checksum ".ruby-version" }}-{{ checksum "Gemfile.lock" }}
+            - bundler-cache-v0-{{ arch }}-{{ checksum ".ruby-version" }}
       - run:
-          name: Install gems and remove unused gems
+          name: Install new gems and remove unused gems
           command: |
             gem install bundler
             bundle check || bundle install
             bundle clean --force
       - save_cache:
-          key: gem-cache-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
+          key: bundler-cache-v0-{{ arch }}-{{ checksum ".ruby-version" }}-{{ checksum "Gemfile.lock" }}
           paths:
             - vendor/bundle
   yarn-install:
     description: Install JavaScript dependencies
     steps:
       - restore_cache:
           name: Restore Yarn cache
-          keys: # Include fallbacks to increase the likeliness of a cache hit
-            - yarn-cache-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}
-            - yarn-cache-{{ arch }}-{{ .Branch }}
-            - yarn-cache-{{ arch }}-
+          keys:
+            - yarn-cache-v0-{{ arch }}-{{ checksum "yarn.lock" }}
+            - yarn-cache-v0-{{ arch }}
       - run:
-          name: Install JS dependencies
+          name: Yarn install
           command: yarn install --cache-folder .yarn-cache
       - save_cache:
-          key: yarn-cache-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}
+          key: yarn-cache-v0-{{ arch }}-{{ checksum "yarn.lock" }}
```

Comment:
This brings up an interesting question: how does one prune the yarn cache folder? The yarn documentation says that `yarn install` will [automatically prune extraneous packages](https://classic.yarnpkg.com/en/docs/cli/prune/), but I am pretty sure it is talking about extraneous packages in `node_modules`, not the yarn cache.

The yarn cache on my Mac is 1.6GB. 😮 

```
$ du -sh ~/Library/Caches/Yarn/v6
1.6G	/Users/mattb/Library/Caches/Yarn/v6
```

Perhaps as a separate PR we can change our CircleCI caching strategy to cache `node_modules` instead of `.yarn-cache`, since the former is automatically pruned and the latter is not.

Reply:
Great question. For timesheet, the yarn cache has held steady at 36M, so I didn't look in to it.

Reply:
OK, turns out the yarn cache is smaller than node_modules in other projects I checked, so clearly there is more going on here that I don't understand.